packages/checkout-common/src/utils.ts

```diff
@@ -1,3 +1,4 @@
 export function assertUnreachable(value: never): never {
+  // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
   throw new Error(`Unexpected case: ${value}`);
```

Comment:
Why can't we do this?
```suggestion
  throw new Error(`Unexpected case: ${value as string}`);
```

Reply:
We can but why is that better? `value` is not a string, it could be anything.

Reply:
You're right, that just avoids the problem, maybe we should provide a default if `value` is not a string - I think this ESLint rule makes sense, as we don't want to have `object [Object]` in an error 😅